### PR TITLE
Configuration Updates

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#11123

*Description of changes:*
* Updated design/styling for Legal Obligations
* Added configuration for additional entities and properties
* Update parameterized unit testing for affected component
* Removed business name from ToDo list items and unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/LegalObligation.vue [hunk 1]
```diff
@@ -1,43 +1,33 @@
...
-                registered or records office addresses as they occur.
+              <li v-for="(changes, index) in getObligations.includedChanges" :key="index" class="obligations-copy">
+                <span class="font-weight-bold">{{changes.label}}</span> - {{changes.description}}
               </li>
             </ul>
           </div>
-          <div class="read-more-line">
-            <span>Find more detailed information <a :href="detailInfoURL" target="_blank">here</a></span>
+          <div class="pt-2">
+            <a :href="getObligations.detailInfoURL" target="_blank">Find out more detailed information</a>
```
Comment (cameron-freshworks) [inline src/components/Dashboard/LegalObligation.vue:26]: For some reason sonar is saying this is a security issue? Any thoughts on this?

Binding a computer property to a url prop is not new, so i'm not sure 🤷

**Comment by cameron-freshworks** [comment]
This work leverages the current configuration structure that existed. 
This could always be expanded/divided in future commits. The goal of this UI ticket was to configure the LegalObligations specifically 👍

**Review by lekshmimallika-aot — APPROVED** [review]

File: src/components/Dashboard/LegalObligation.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/Dashboard/LegalObligation.vue:26]: Maybe it's not happy about the lack of type for `detailInfoURL` ?

I don't think there's a security risk here. Are you able to login to Sonarcloud to change this hotspot to "Safe" status? If not, I can do it.

File: src/components/Dashboard/LegalObligation.vue [hunk 2]
```diff
@@ -1,43 +1,33 @@
...
+            <v-icon color="primary" class="ml-1" small>mdi-open-in-new</v-icon>
           </div>
-          <div class="read-more-line">
-            <span class="app-blue cursor-pointer" @click="readMoreFlag = false">Read less...</span>
+          <div class="pt-2">
+            <span class="app-blue cursor-pointer" @click="readMoreFlag = false"><u>Read less</u></span>
```
Comment (severinbeauvais) [inline src/components/Dashboard/LegalObligation.vue:30]: In Create UI, "read more..." and "read less..." are not underlined, but they are rendered as "link buttons". Is there a different design here?

image

**Review by severinbeauvais — COMMENTED** [review]
Good job on parameterizing some tests 👍

File: src/components/Dashboard/LegalObligation.vue (hunk 2 above)
Comment (cameron-freshworks) [reply src/components/Dashboard/LegalObligation.vue:30]: o wow, yea that is interesting.. definitely different design 🤔 
https://projects.invisionapp.com/d/main/#/console/21712216/460798286/preview

These designs show this 'expand' btn with underline, without the `...` etc

Screen Shot 2022-02-14 at 2 56 48 PM

File: src/components/Dashboard/LegalObligation.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/Dashboard/LegalObligation.vue:26]: Interesting, i will check it out!

**Comment by cameron-freshworks** [comment]
> Good job on parameterizing some tests 👍

Thank you! 
Enjoyable task when we have the time for them :)

File: src/components/Dashboard/LegalObligation.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/Dashboard/LegalObligation.vue:30]: OK, not our problem.

File: src/components/Dashboard/LegalObligation.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/Dashboard/LegalObligation.vue:26]: Added an interface 🤞

Comment (cameron-freshworks) [reply src/components/Dashboard/LegalObligation.vue:26]: Looks like it still needs the Sonar Approval, i'm not sure i can update this.
Logged in, however it still said i required permissions.

Comment (severinbeauvais) [reply src/components/Dashboard/LegalObligation.vue:26]: OK, I updated this:

image

**Review by severinbeauvais — APPROVED** [review]
Now Sonarcloud is complaining about code duplication, but I think there's no easy way to fix this, so GTG.

File: src/components/Dashboard/LegalObligation.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/Dashboard/LegalObligation.vue:26]: Thank you!

**Pull request merged**